### PR TITLE
Consumer error handling

### PR DESCRIPTION
**Pull request by gchazot**
* Cleans up the exception handling in the consumer to handle different problems separately
* Avoids the logging-loop-of-death when an error happens in rabbit client / `.get()`
* Adds a few perks for debugging

TODO:
* [x] Add handling of reconnection on Rabbit errors

### CODE REVIEW

File: queue_util/consumer.py [hunk 1]
```diff
@@ -149,52 +149,43 @@ def run_forever(self, wait_timeout_seconds=None, **kwargs):
...
 
-                message = self.source_queue.get(block=True, timeout=wait_timeout_seconds)
-                data = message.payload
+                try:
+                    message = self.source_queue.get(block=True, timeout=wait_timeout_seconds)
+                    data = message.payload
+                except queue.Empty:
+                    continue
+                except Exception as e:
+                    logging.exception("Exception getting message from Rabbit", str(e))
```
Comment (Fizzadar) [inline queue_util/consumer.py:158]: Think this needs a `%s` to work (tested in py3.8):

```suggestion
                    logging.exception("Exception getting message from Rabbit: %s", str(e))
```

**Review by Fizzadar — APPROVED** [review]
LGTM!, just one little thing

File: queue_util/consumer.py (hunk 1 above)
Comment (gchazot) [reply queue_util/consumer.py:158]: Oh that's how it works! :facepalm:

Comment (Fizzadar) [reply queue_util/consumer.py:158]: I totally thought it'd work but wasn't 100% and actually errored out locally!

Comment (gchazot) [reply queue_util/consumer.py]: 9dcb155

**Review by Fizzadar — APPROVED** [review]
Updates LGTM!

**Pull request merged**